server/notification-providers/nostr.js

```diff
@@ -0,0 +1,108 @@
+const NotificationProvider = require("./notification-provider");
+global.crypto = require("crypto");
```

Comment:
Why is this line included?
```suggestion
```

Reply:
Without it, on Node 18 there is an error from the nostr-tools module that crypto is not defined. Node 19+ doesn't need it.  There is a closed issue for nostr-tools about it: https://github.com/nbd-wtf/nostr-tools/issues/192

I see they have a better method for Node 18 compatibility now, but I'm not sure how to implement it:
```
// node.js 18 and earlier requires globalThis.crypto polyfill.
import { webcrypto } from 'node:crypto';
// @ts-ignore
if (!globalThis.crypto) globalThis.crypto = webcrypto;
```
Here is the error with Node 18:
```
ReferenceError: crypto is not defined
    at Object.encrypt (/home/ansible/uptime-kuma/node_modules/nostr-tools/lib/nostr.cjs.js:994:19)
    at Nostr.send (/home/ansible/uptime-kuma/server/notification-providers/nostr.js:34:44)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Socket.<anonymous> (/home/ansible/uptime-kuma/server/server.js:1209:27)
```

Reply:
Please include a comment that this is a `polyfill for node <=  18`